### PR TITLE
Add email validation/parsing on incoming NewCertificate requests.

### PR DESCRIPTION
**Pull request by d1str0**
### Context
This PR addresses the lack of validation of user input on email addresses for Certificates.

### Approach

We use net/mail to ParseAddress(), check for errors, and if no errors arise, return a valid certificate.
https://golang.org/pkg/net/mail/#ParseAddress

### Testing
No tests added, and no manual testing. Another PR is currently addressing Unit tests for Certificate but that change is not included in this PR. :(

### Misc.

Still need to validate domains submitted and also refactor this app logic out of the model and into the service.

### CODE REVIEW

**Comment by d1str0** [comment]
@mklepp @skoganti-iws  @gtoveyiws 

This PR uses a template. The template can be found here: 
https://github.com/ImageWare/TLSential/blob/develop/.github/pull_request_template.md

Here is the template:

## This PR addresses the following issues: 


### Context

In one or two sentences, what problem is this PR trying to solve?

### Approach

Briefly, how does this PR solve the issue?

### Testing

What was your strategy for testing the new code?

### Misc.

Screenshots? Loose ends? Concerns?

**Review by debus — DISMISSED** [review]
LGTM

**Comment by debus** [comment]
Btw I really like the template for PRs. I think it will really help to easily communicate what each PR is doing and why. As well as force people to consider all aspects of the PR before submitting (e.g. If the brief description gets too long they may consider breaking it into smaller chunks. If they totally blank out at the Testing Strategy they may remember unit tests are a thing lol).

**Comment by brooks42** [comment]
FWIW I actually prefer to go a step further and name branches `[dev initials]/[Jira ticket]-[short description]`, for example `cb/gi-994-enroll-text-fix`. Then in PR titles I specifically place affected Jira tickets in the name so we can more easily search for PRs that changed the code for related issues. Just an extra facet to consider.

**Review by brooks42 — DISMISSED** [review]
Fix merge conflicts and LGTM 👍

**Comment by debus** [comment]
> FWIW I actually prefer to go a step further and name branches `[dev initials]/[Jira ticket]-[short description]`, for example `cb/gi-994-enroll-text-fix`. Then in PR titles I specifically place affected Jira tickets in the name so we can more easily search for PRs that changed the code for related issues. Just an extra facet to consider.

Although in this Repo's case we'd want to reference Github Issues instead of JIRA

File: model/certificate.go [hunk 1]
```diff
@@ -66,11 +66,9 @@ type Certificate struct {
 	ACMEKey          *ecdsa.PrivateKey
 }
 
-var ErrInvalidDomains = errors.New("invalid domains")
-var ErrInvalidEmail = errors.New("invalid email")
```
Comment (debus) [inline model/certificate.go]: Wait so are these errors gone? Aren't they being referenced somewhere else?

File: model/certificate.go [hunk 2]
```diff
@@ -96,7 +94,8 @@ func NewCertificate(domains []string, email string) (*Certificate, error) {
 		Secret:     secret,
 		Domains:    domains,
 		CommonName: common,
-		ACMEEmail:  e.Address,
+		RenewAt:    DefaultRenewAt,
+    ACMEEmail:  e.Address,
```
Comment (debus) [inline model/certificate.go]: Nitpick but the merge seems to have messed up the tab-scoping

**Comment by d1str0** [comment]
Branch names and PR titles can't really be templated or enforced. Although I don't disagree with adding some more standards to branch names and titles, as well as commit names, etc.

(that and tagging all versions)

**Review by debus — APPROVED** [review]

**Comment by brooks42** [comment]
> > FWIW I actually prefer to go a step further and name branches `[dev initials]/[Jira ticket]-[short description]`, for example `cb/gi-994-enroll-text-fix`. Then in PR titles I specifically place affected Jira tickets in the name so we can more easily search for PRs that changed the code for related issues. Just an extra facet to consider.
> 
> Although in this Repo's case we'd want to reference Github Issues instead of JIRA

Right :P

File: model/certificate.go (hunk 2 above)
Comment (d1str0) [reply model/certificate.go]: the in-browser merge conflict editor isn't great about tabs vs spaces. :P

File: model/certificate.go (hunk 1 above)
Comment (d1str0) [reply model/certificate.go]: They were just moved upwards in the file.

**Pull request merged**